packages/common/src/Utils.ts

```diff
@@ -476,3 +476,34 @@ export async function waitForSuccess<T> (
     }
   })
 }
+
+/**
+ * From https://ethereum.stackexchange.com/a/43649
+ * May be outdated but should do the job
+ */
+export function getProviderImplementation (provider: any): string {
```

Comment:
Do we need this? Determine the type of ropc provider is unrelated to the issue of supporting ethers.

Reply:
Not really. Just tried to check as many known providers as possible so we only log the warning for ones that are not recognized at all. Can be simplified or even removed.